[UWP] Fixes to Cell for ListView and CollectionView

CliffAgius opened the pull request:
### Description of Change ###

Fix for #10473 that is a massive blocker for UWP development using List and Collection views in XF.  This fix was designed by @gentledepp and I am putting it forward as a PR to the main XF repo as they had only made the changes to thier company fork.  I have tested these changes and they seem to fix the issue on the project we are building.

### Issues Resolved ### 
- fixes #10473 

### API Changes ###
None

### Platforms Affected ### 
- UWP

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###
<!-- To be completed by reviewers -->

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

rmarinho commented on azure-pipelines.yml:
```diff
@@ -27,34 +27,6 @@ resources:
       name: xamarin/yaml-templates
       endpoint: xamarin
 
-trigger:
```
please remove these changes

CliffAgius replied:
Odd I havn't touched that file... Ok will check and update standby.

CliffAgius commented:
here you go @rmarinho I reverted the changes I have no idea how or why that file was changed but it's back to how it was and only the one file is now changed which is the one that should I hope fix the UWP Memory leak.

jfversluis commented:
/azp run
